dsp-shared/src/test/scala/dsp/valueobjects/SchemaSpec.scala

```diff
@@ -0,0 +1,483 @@
+      val guiElement    = guiElementSlider
+      val guiAttributes = scala.collection.immutable.List(guiAttributeMin, guiAttributeMax)
+      val result        = Schema.validateGuiObjectSlider(guiElement, guiAttributes)
+      assertTrue(
+        result == Validation.succeed(guiAttributes)
+      )
+    },
+    test(
+      "pass gui element 'salsah-gui#Slider' with too many gui attributes 'min=1','max=10', and 'min=80' and return an error"
+    ) {
+      val guiElement    = guiElementSlider
+      val guiAttributes = scala.collection.immutable.List(guiAttributeMin, guiAttributeMax, guiAttributeSize)
+      assertTrue(
+        Schema.validateGuiObjectSlider(guiElement, guiAttributes) == Validation.fail(
+          BadRequestException(
+            "Wrong number of gui attributes. salsah-gui:guiElement GuiElement(http://www.knora.org/ontology/salsah-gui#Slider) needs 2 salsah-gui:guiAttribute 'min' and 'max', but found 3: List(GuiAttribute(min,1), GuiAttribute(max,10), GuiAttribute(size,80))."
+          )
+        )
+      )
+    },
+    test(
+      "pass gui element 'salsah-gui#Slider' with too many gui attributes 'min=1','hlist=http://rdfh.ch/lists/082F/PbRLUy66TsK10qNP1mBwzA' and return an error"
+    ) {
+      val guiElement    = guiElementSlider
+      val guiAttributes = scala.collection.immutable.List(guiAttributeSize)
+      assertTrue(
+        Schema.validateGuiObjectSlider(guiElement, guiAttributes) == Validation.fail(
+          BadRequestException(
+            "Wrong number of gui attributes. salsah-gui:guiElement GuiElement(http://www.knora.org/ontology/salsah-gui#Slider) needs 2 salsah-gui:guiAttribute 'min' and 'max', but found 1: List(GuiAttribute(size,80))."
+          )
+        )
+      )
+    }
+  )
+
+  private val guiObjectTest = suite("gui object")(
+    test(
+      "pass valid gui element with duplicated gui attributes and return an error"
+    ) {
+      assertTrue(
+        Schema.GuiObject
+          .make(
+            scala.collection.immutable.List(guiAttributeMin, guiAttributeMin),
+            Some(guiElementSlider)
+          ) == Validation.fail(
+          BadRequestException(
+            "Duplicate gui attributes for salsah-gui:guiElement Some(GuiElement(http://www.knora.org/ontology/salsah-gui#Slider))."
+          )
+        )
+      )
+    }
+  )
+
+  private val guiObjectListTest = suite("gui object - List")(
+    test(
+      "pass gui element 'salsah-gui#List' with gui attribute 'hlist' and successfully create value object"
+    ) {
+      (for {
+        guiObject <- Schema.GuiObject
+                       .make(
+                         scala.collection.immutable.List(guiAttributeHlist),
+                         Some(guiElementList)
+                       )
+      } yield assertTrue(
+        guiObject.guiAttributes == scala.collection.immutable.List(guiAttributeHlist)
+      ) &&
+        assertTrue(
+          guiObject.guiElement == Some(guiElementList)
+        )).toZIO
+    },
```

Comment:
I don't understand why we need to use a for comprehension and toZIO here (and in the following suites) but didn't in the first one

Reply:
Like this, the `.fold(e => throw e.head, v => v)` is not needed to get to the actual value object, instead everything is treated inside the for comprehension. But now that you mentioned it, I changed it to the `.fold(...)`. I think it's easier to understand.